### PR TITLE
fix: routing modifications

### PR DESCRIPTION
**Pull request by sivaRajanaDivami**
# TITLE = Updated names in Breadcrumb and routing from the breadcrumb

## Description
Summary of changes

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

## Screenshots
Screenshot 2022-09-12 at 10 04 36 PM
Screenshot 2022-09-12 at 10 04 49 PM

### CODE REVIEW

File: app/cdap/components/GridTable/components/Breadcrumb/index.tsx [hunk 1]
```diff
@@ -22,16 +22,18 @@ import React from 'react';
 import { getCurrentNamespace } from 'services/NamespaceStore';
 import { Link } from 'react-router-dom';
 
-const BreadCrumb = ({ datasetName }) => {
+const BreadCrumb = ({ datasetName, location }) => {
   const classes = useStyles();
+  const sourcePath =
+    location.state.from === 'Workspaces' ? 'home' : `datasources/${location.state.path}`;
```
Comment (adityadivami) [inline app/cdap/components/GridTable/components/Breadcrumb/index.tsx]: can we move strings to a constant file

// constants.ts
export const WORKSPACE_URL_PATH = "workspaces"
export const HOME_URL_PATH = "home"

Try to do this across the changes in this PR only for now and follow for all PRs going forward

File: app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx [hunk 2]
```diff
@@ -90,8 +92,16 @@ const TabLabelCanSample = ({
...
 
+  const indexOfSelectedDataset = location.pathname.lastIndexOf('/');
+  const requiredPath = location.pathname.slice(indexOfSelectedDataset + 1);
+
   return workspaceId ? (
-    <Redirect to={`/ns/${getCurrentNamespace()}/wrangler-grid/${workspaceId}`} />
+    <Redirect
+      to={{
+        pathname: `/ns/${getCurrentNamespace()}/wrangler-grid/${workspaceId}`,
+        state: { from: 'Data Sources', path: requiredPath },
```
Comment (adityadivami) [inline app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx]: same as below comment, move strings "Data Sources" to constants.ts specific to this component like index.tsx, styles.ts, utils.ts and constants.ts

**Review by adityadivami — CHANGES_REQUESTED** [review]
please update the review comments

File: app/cdap/components/GridTable/components/Breadcrumb/index.tsx (hunk 1 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/GridTable/components/Breadcrumb/index.tsx]: Created constants.ts file and transferred all the constants to it.

File: app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx (hunk 2 above)
Comment (sivaRajanaDivami) [reply app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx]: created constants.ts file and moved all the constants to it from index.tsx file.

File: app/cdap/components/ConnectionList/Components/TabLabelCanSample/constants.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const DATASOURCES_LABEL = 'Data Sources';
```
Comment (njbriggs) [inline app/cdap/components/ConnectionList/Components/TabLabelCanSample/constants.ts:17]: Copyright header

File: app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx [hunk 4]
```diff
@@ -101,7 +112,7 @@ const TabLabelCanSample = ({
         <div onClick={() => onExplore(entity)}>
           <Box className="wranglingHover">
             <WrangelIcon />
```
Comment (seanfeld) [inline app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx]: Is this misspelled? should it be `WrangleIcon`?

Comment (richadivami) [reply app/cdap/components/ConnectionList/Components/TabLabelCanSample/index.tsx]: Yes, it was a typo error, corrected it

File: app/cdap/components/ConnectionList/Components/TabLabelCanSample/constants.ts (hunk 3 above)
Comment (richadivami) [reply app/cdap/components/ConnectionList/Components/TabLabelCanSample/constants.ts:17]: Added

File: app/cdap/components/GridTable/components/Breadcrumb/constants.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const MATCH_SOURCE = 'Workspaces';
```
Comment (njbriggs) [inline app/cdap/components/GridTable/components/Breadcrumb/constants.ts:1]: Copyright header

File: app/cdap/components/WrangleHome/Components/OngoingDataExploration/constants.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export const WORKSPACES_LABEL = 'Workspaces';
```
Comment (njbriggs) [inline app/cdap/components/WrangleHome/Components/OngoingDataExploration/constants.ts:1]: Copyright header

**Pull request closed without merging**